components/org.wso2.carbon.identity.oauth.ciba/src/main/java/org/wso2/carbon/identity/oauth/ciba/api/CibaAuthServiceImpl.java

```diff
@@ -0,0 +1,172 @@
+        Timestamp issuedTime = new Timestamp(issuedTimeInMillis);
+        long expiryTime = getExpiresIn(cibaAuthCodeRequest);
+        String[] scopes = cibaAuthCodeRequest.getScopes();
+        cibaAuthCodeDO.setCibaAuthCodeKey(generateAuthCodeKey());
+        cibaAuthCodeDO.setAuthReqID(generateAuthRequestId());
+        cibaAuthCodeDO.setConsumerKey(cibaAuthCodeRequest.getIssuer());
+        cibaAuthCodeDO.setIssuedTime(issuedTime);
+        cibaAuthCodeDO.setLastPolledTime(issuedTime); // Initially last polled time is set to issued time.
+        cibaAuthCodeDO.setAuthReqStatus(AuthReqStatus.REQUESTED);
+        cibaAuthCodeDO.setInterval(CibaConstants.INTERVAL_DEFAULT_VALUE_IN_SEC);
+        cibaAuthCodeDO.setExpiresIn(expiryTime);
+        cibaAuthCodeDO.setScopes(scopes);
+        return cibaAuthCodeDO;
+    }
+
+    /**
+     * Builds and returns CibaAuthCodeResponse.
+     *
+     * @param cibaAuthCodeDO      DO with information regarding authenticationRequest.
+     * @param cibaAuthCodeRequest Auth Code request object.
+     * @throws CibaCoreException Exception thrown from CibaCore Component.
+     */
+    private static CibaAuthCodeResponse buildAuthCodeResponse(CibaAuthCodeRequest cibaAuthCodeRequest,
+                                                              CibaAuthCodeDO cibaAuthCodeDO) throws CibaCoreException {
+
+        String clientID = cibaAuthCodeRequest.getIssuer();
+        try {
+            CibaAuthCodeResponse cibaAuthCodeResponse = new CibaAuthCodeResponse();
+            String user = cibaAuthCodeRequest.getUserHint();
+            OAuthAppDO appDO = OAuth2Util.getAppInformationByClientId(clientID);
+            String callbackUri = appDO.getCallbackUrl();
+            cibaAuthCodeResponse.setAuthReqId(cibaAuthCodeDO.getAuthReqID());
+            cibaAuthCodeResponse.setCallBackUrl(callbackUri);
+            cibaAuthCodeResponse.setUserHint(user);
+            cibaAuthCodeResponse.setClientId(clientID);
+            cibaAuthCodeResponse.setScopes(OAuth2Util.buildScopeString(cibaAuthCodeRequest.getScopes()));
+            cibaAuthCodeResponse.setExpiresIn(cibaAuthCodeDO.getExpiresIn());
+
+            if (StringUtils.isNotBlank(cibaAuthCodeRequest.getBindingMessage())) {
+                cibaAuthCodeResponse.setBindingMessage(cibaAuthCodeRequest.getBindingMessage());
+            }
+
+            if (StringUtils.isNotBlank(cibaAuthCodeRequest.getTransactionContext())) {
+                cibaAuthCodeResponse.setTransactionDetails(cibaAuthCodeRequest.getTransactionContext());
+            }
+
+            if (log.isDebugEnabled()) {
+                log.debug("Successful in creating AuthCodeResponse for the client: " + clientID);
+            }
+            return cibaAuthCodeResponse;
+        } catch (IdentityOAuth2Exception | InvalidOAuthClientException e) {
+            throw new CibaCoreException("Error in creating AuthCodeResponse for the client: " + clientID, e);
```

Comment:
Would be better to identify client exceptions and throw a CIBAClientException.

For example, InvalidOAuthClientException is thrown if the application does not exist. In such cases we can wrap it an throw a client exception so that the endpoint will not log those errors.

Reply:
Modified.